### PR TITLE
parser: cache variable in parser to avoid allocation

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

**Comment by zimulala** [comment]
LGTM

File: parser/yy_parser.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ type Parser struct {
 	cache     []yySymType
 	src       string
 	lexer     Scanner
+
+	yylval yySymType
```
Comment (shenli) [inline parser/yy_parser.go]: Add comments for this two member variables.

**Comment by shenli** [comment]
Any performance improvement?

**Comment by tiancaiamao** [comment]
not obvious, but it reduce allocation.

```
BenchmarkParse-4	   30000	     53280 ns/op	    5888 B/op	     112 allocs/op
BenchmarkParse-4	   30000	     53884 ns/op	    5600 B/op	     106 allocs/op
```

**Comment by shenli** [comment]
LGTM

**Pull request merged**